rsmtool/utils.py

```diff
@@ -667,6 +674,120 @@ def get_output_directory_extension(directory, experiment_id):
     return extension
 
 
+class Thumbnail:
```

Comment:
The actual technique for generating the thumbnails works really well but I have some suggestions for the semantics of this class:

1. I would actually modify the  `__init__()` method to take the path to the image file and then do the checks that the image exists, get its relative path, and then store the path in the `image_path` attribute (along with setting `current_id` to 0). This makes it much clearer when creating a `Thumbnail` instance in user land. 

2. I would rename `convert_to_html()` to just be `to_html()` to follow other libraries' conventions.

3. I would rename `convert` to `show()`.

4. Right now, there's no indication that the thumbnail is clickable. I would add `cursor: pointer` to the CSS and also add something like "click to enlarge" text below the thumbnail. It would also be nice to have some thing in the top of the report in bold saying that the images have been converted to thumbnails and the user can click on them to enlarge them. We may be sending these reports around to people who aren't that tech savvy.

Reply:
This all makes sense except for the `__init__()` method, which can't take the file path, since I'm only instantiating the object once. I can check that the file exists in the `to_html()` method. Let me know if that sounds all right.